User search implementation

solocommand opened the pull request:
- [x] Adds microservice action to return paginated results based on search terms
  - Existing `matchForApp` does not support pagination, used for autocompleters
- [x] Adds definition/resolver for AppUser
  - Avoided going down the directive route due to needing something to signify the model/service
- [x] Management implementation, make users searchable

GraphQL query/response
<img width="1200" alt="Screen Shot 2019-06-25 at 11 35 12 AM" src="https://user-images.githubusercontent.com/1778222/60116453-a5aeb400-973d-11e9-9740-4fa60b9b38b1.png">

Mgmt UI
<img width="1145" alt="Screen Shot 2019-06-25 at 1 37 35 PM" src="https://user-images.githubusercontent.com/1778222/60124236-ca5f5780-974e-11e9-8904-d488be1c1008.png">
<img width="1147" alt="Screen Shot 2019-06-25 at 1 37 22 PM" src="https://user-images.githubusercontent.com/1778222/60124237-ca5f5780-974e-11e9-9294-bfd87a3818e7.png">
<img width="1146" alt="Screen Shot 2019-06-25 at 1 36 54 PM" src="https://user-images.githubusercontent.com/1778222/60124238-ca5f5780-974e-11e9-9393-55c265e5ad1a.png">
<img width="1147" alt="Screen Shot 2019-06-25 at 1 36 39 PM" src="https://user-images.githubusercontent.com/1778222/60124239-ca5f5780-974e-11e9-8f3b-0e4182fc04d8.png">

solocommand committed:
Suppory query params in findForApplication

solocommand committed:
Abstract buildRegex function

solocommand committed:
Add matchManyForApp common action

solocommand committed:
Add support for paginated user searches

solocommand committed:
Pass sort and pagination through to service

solocommand committed:
Move pagination/sort out of query input

zarathustra323 commented on packages/mongoose-plugins/src/application-plugin.js:
```diff
@@ -13,11 +13,10 @@ module.exports = function applicationPlugin(schema, { options = {}, collateWhen
-    const query = { applicationId };
     return this.paginate({
-      query,
+      query: { applicationId, ...query },
```
This needs to be reversed, otherwise the `applicationId` can be overridden by the query

zarathustra323 commented on packages/utils/src/actions/index.js:
```diff
@@ -12,6 +13,7 @@ module.exports = {
   listForApp,
   listForOrg,
   matchForApp,
+  matchManyForApp,
```
What's the distinction here? `matchForApp` and `matchManyForApp` both return multiple results. These should be consolidated.

solocommand committed:
Port list-controls

zarathustra323 commented on packages/utils/src/actions/match-many-for-app.js:
```diff
@@ -0,0 +1,20 @@
+  const sort = { [field]: 1 };
+
+  return Model.findForApplication(applicationId, fields, { sort, query, pagination });
+};
```
Consolidate this logic into `matchForApp`

zarathustra323 commented on packages/utils/src/index.js:
```diff
@@ -1,10 +1,12 @@
 
 module.exports = {
   actions,
+  buildRegex,
```
Shouldn't need to put this here anymore...

solocommand committed:
Support searching for AppUsers

zarathustra323 reviewed (changes requested):
"Fix" the match input group controls so that the left-side button (next to the input) is not rounded. Also, the input itself needs drop shadow added to it to the match the buttons

zarathustra323 reviewed (changes requested):
Also, we no longer need the `Show: 24` button (nor the limit param) as these pages are now infinite scroll

solocommand committed:
Remove buildRegex export

solocommand committed:
Do not allow Application ID to be overwritten

solocommand committed:
Remove limit/limitOptions

solocommand committed:
Remove right border radius from button

solocommand committed:
Add box shadow to search box

solocommand commented:
Removed limit, fixed box shadow and border radius
<img width="1036" alt="Screen Shot 2019-06-25 at 2 27 19 PM" src="https://user-images.githubusercontent.com/1778222/60127293-6db36b00-9755-11e9-83e1-132519bf97df.png">

solocommand committed:
Unify matchForApp/matchManyForApp

zarathustra323 commented on packages/utils/src/actions/match-for-app.js:
```diff
@@ -23,16 +8,22 @@ module.exports = async (Model, {
-  return Model.find(query, fields, { sort });
+  const defaultSort = { field, order: 'desc' };
+  const payload = {
+    sort: { ...defaultSort, sort },
```
Spread `...sort`

zarathustra323 commented on services/graphql/src/graphql/definitions/app-user.js:
```diff
@@ -3,11 +3,12 @@ const gql = require('graphql-tag');
 
 extend type Query {
-  appUsers(input: AppUsersQueryInput!): AppUserConnection! @requiresAppRole
+  appUsers(pagination: PaginationInput = {}, sort: AppUserSortInput = {}): AppUserConnection! @requiresAppRole
```
The `pagination` and `sort` args should be combined into a single input. See
https://github.com/base-cms/id-me/blob/1d512f268b539fdde983aace151655810edb7dd6/services/graphql/src/graphql/definitions/team.js#L7
And
https://github.com/base-cms/id-me/blob/1d512f268b539fdde983aace151655810edb7dd6/services/graphql/src/graphql/definitions/team.js#L60

zarathustra323 commented on services/graphql/src/graphql/definitions/app-user.js:
```diff
@@ -3,11 +3,12 @@ const gql = require('graphql-tag');
   activeAppUser: AppUser @requiresAuth(type: AppUser)
   activeAppContext: AppContext! @requiresApp # must be public
   checkContentAccess(input: CheckContentAccessQueryInput!): AppContentAccess! @requiresApp # must be public
+  matchAppUsers(input: MatchAppUsersQueryInput!, pagination: PaginationInput = {}, sort: AppUserSortInput = {}): AppUserConnection! @requiresAppRole
```
Same as above.

zarathustra323 commented on services/graphql/src/graphql/resolvers/access-level.js:
```diff
@@ -35,21 +35,22 @@ module.exports = {
      *
      */
-    matchAccessLevels: (_, { input }, { app }) => {
+    matchAccessLevels: async (_, { input }, { app }) => {
```
You might as well convert this to use a `AccessLevelConnection` so it's consistent with everything else.

solocommand committed:
Move buildRegex back into matchForApp

solocommand committed:
Spread sort

solocommand committed:
Revert "Move pagination/sort out of query input"

This reverts commit fcaae7c3fb6fb712c7212f6ba9bdb3c24371e39e.

solocommand committed:
Update manage GQL

solocommand committed:
Oops, update matchAppUsers definition

solocommand committed:
Convert access levels to use the connection

zarathustra323 committed:
Rework button styling/grouping

zarathustra323 committed:
Adjust search input control z-index

zarathustra323 committed:
Use new component template syntax

zarathustra323 committed:
Adjust input group btn last child styles

zarathustra323 committed:
Do not disable sort on search

zarathustra323 committed:
Hide x button on input search fields

zarathustra323 committed:
Add trailing commas

zarathustra323 committed:
Remove isSortDisabled computed prop

zarathustra323 committed:
Add ability to clear/reset list controls

zarathustra323 committed:
Change label to reset

zarathustra323 commented:
Default view:
![image](https://user-images.githubusercontent.com/3289485/60455734-e0609280-9bfc-11e9-8fc2-66f3af68443c.png)

Search phrase view:
![image](https://user-images.githubusercontent.com/3289485/60455752-ece4eb00-9bfc-11e9-8fac-267a91ae06c2.png)

Search phrase focused view (no Chrome X button):
![image](https://user-images.githubusercontent.com/3289485/60455766-f8d0ad00-9bfc-11e9-8fda-57c852b13e6d.png)
